### ISSUE TITLE
use classImplements to have ReactorBase conditionally implement IReactorThreads

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#11071]
|[<img alt="mwhudson's avatar" src="https://avatars.githubusercontent.com/u/672022?s=50" width="50" height="50">](https://github.com/mwhudson)| @mwhudson reported|
|-|-|
|Trac ID|trac#1532|
|Type|enhancement|
|Created|2006-03-02 16:42:28Z|

The point of this excercise is that fairly soon docextractor will be able to analyze the classImplements call and "implements(*_impllist)" is likely to always confuse it.

Attachments:

* [t.i.base.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/00b/00b4a051351b585e1f29ffce2b7ff5d29d629966/8717ece6edbdee6841d762853d738cc96a78d683.diff) (1002 bytes) - added by mwh on 2006-03-02 16:43:08Z - the diff

<details><summary>Searchable metadata</summary>

```
trac-id__1532 1532
type__enhancement enhancement
reporter__mwh mwh
priority__low low
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1141317748000000 1141317748000000
changetime__1141326559000000 1141326559000000
version__ 
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set status to `closed`|
|-|-|

(In [16118]) Apply mwh's patch to simplify implements declaration on ReactorBase

Fixes [#11071](https://github.com/twisted/twisted/issues/11071)

This makes it easier for docextractor to figure out, statically, what is going on.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
